_posts/2017-12-27-cheap-docker-swarm-cluster-aws-terraform.md

```diff
@@ -0,0 +1,98 @@
+
+## Self Hosting with Commodity Hardware
+
+So let’s start with our baseline.  What would it cost to get up and running with a 3 node cluster using generally available?  Just using parts from NewEgg, you can build a cheap commodity system for about $500.  Circa 2017, this would get you 4GB of RAM, and a 2.9GHz Dual Core Intel processor.  Most of that cost is overhead, such as power supply, case, motherboard, storage space, networking and cooling.  To make a good comparison with the cloud, let’s amortize the cost of the hardware over a 2 year period, which gives us an amortized rate of $20.83/month.
+
+But we also need to consider the cost of electricity.  Let’s assume a $0.10/kWh rate, which is quite inexpensive, but also common across large parts of North America.  Assuming a 300W constant power draw, we are looking at $21.60/month, just for electricity.  Our cost of electricity is about the same amount as our amortized hardware costs!  Not really what I had in mind for cheap, but this is just our baseline.
+
+Ignoring a large number of other considerable factors, this gives us an estimated monthly cost of operation of $42.43.  This is the amortized cost of the hardware, plus the cost of electricity.
+
+There are a number of alternatives we could explore on the self hosted side that could reduce our costs.  For example, we could explore the use of low power ARM processors, which are both inexpensive as well as efficient.  Depending on the intended purpose, this may be a valid option.  But for the purpose of comparison, we will stick with an Intel processor.
+
+To summarize:
+
+| Description | Cost |
+| ----------- | ---- |
+| Monthly Amortized Hardware Costs | $20.83 |
+| Monthly Electricity Usage        | $21.60 |
+| Total                            | $42.43 |
+
+## Cloud Hosting
+
+With [AWS](https://aws.amazon.com/), there are a large number of server types to choose from.  A comparable EC2 type to our self-hosted solution would be the t2.medium instance.  [Here’s a great site that compares the various options and prices.](https://www.ec2instances.info/)  If we use on-demand pricing, we will get higher availability, but we are paying full price.  We have the option to go with spot pricing and we are going to sacrifice availability for some savings.  Here’s the comparison for the t2.medium, with current 2017 pricing.
+
+
+| Lifecycle | Total Hourly Cost | Total Monthly Cost |
+| --------- | ----------------- | ------------ |
+| On-Demand | $0.0464 | $33.41 |
+| Spot | $0.0185 | $13.32 |
+
+Even with full availability, we are still getting cheaper prices than our self-build, but a significant savings when we look at spot pricing.
+
+## Setting it all up
+
+Another major factor we need to discuss is how long it takes to set this stuff up.  We’ve done a simple comparison which makes a compelling case for the cloud.  But where the cloud really shines is in the time savings setting the cluster up.
```

Comment:
Have we really made a compelling case for the cloud? All we've done is said "the cloud saves us money." If that's the one argument, why do people NOT do this? Stop assuming I agree with your reasoning and SHOW ME YOUR REASONING. I'm not an idiot; you don't have to exhaust me with logic. Just ... say why you think what you think.